cmd/geth/main.go

```diff
@@ -325,3 +319,32 @@ func startNode(ctx *cli.Context, stack *node.Node) {
 		}
 	}
 }
+
+// unlockAccounts unlocks any account specifically requested.
+func unlockAccounts(ctx *cli.Context, stack *node.Node, cfg gethConfig) {
+	// personalExposed checks whether personal API is exposed to http.
+	personalExposed := func() bool {
+		for _, module := range cfg.Node.HTTPModules {
+			if module == "personal" {
+				return true
+			}
+		}
+		return false
+	}
+
+	// If user explicitly enable insecure account unlocking or account-related APIs will
+	// not be exposed by http, execute unlock operation.
+	enable := ctx.GlobalBool(utils.InsecureUnlockFlag.Name)
```

Comment:
nit: I would rename enable to something like isInsecureFlagEnabled

then invert the condition for better readability.

if  !isInsecureFlagEnabled && personalExposed() {
 log
 return
}